tests/checks/integration/test_docker_daemon.py

```diff
@@ -0,0 +1,439 @@
+            ('docker.image.size', ['image_name:redis', 'image_tag:latest']),
+            ('docker.image.size', ['image_name:nginx', 'image_tag:1', 'image_tag:1.9', 'image_tag:1.9.4', 'image_tag:latest']),
+            ('docker.image.size', ['image_name:nginx', 'image_tag:1.9.0']),
+            ('docker.image.size', ['image_name:nginx', 'image_tag:1.7.11']),
+            ('docker.image.size', ['image_name:nginx', 'image_tag:1.9.2']),
+            ('docker.image.size', ['image_name:nginx', 'image_tag:1.9.3']),
+            ('docker.image.virtual_size', ['image_name:nginx', 'image_tag:1.9.1']),
+            ('docker.image.virtual_size', ['image_name:nginx', 'image_tag:1.7', 'image_tag:1.7.12']),
+            ('docker.image.virtual_size', ['image_name:nginx', 'image_tag:1.9.0']),
+            ('docker.image.virtual_size', ['image_name:nginx', 'image_tag:1.7.11']),
+            ('docker.image.virtual_size', ['image_name:nginx', 'image_tag:1', 'image_tag:1.9', 'image_tag:1.9.4', 'image_tag:latest']),
+            ('docker.image.virtual_size', ['image_name:nginx', 'image_tag:1.9.2']),
+            ('docker.image.virtual_size', ['image_name:nginx', 'image_tag:1.9.3']),
+            ('docker.image.virtual_size', ['image_name:redis', 'image_tag:latest']),
+            ('docker.images.available', None),
+            ('docker.images.intermediate', None),
+            ('docker.mem.cache', ['container_name:test-new-nginx', 'docker_image:nginx', 'image_name:nginx']),
+            ('docker.mem.cache', ['container_name:test-new-redis-latest', 'docker_image:redis:latest', 'image_name:redis', 'image_tag:latest']),
+            ('docker.mem.rss', ['container_name:test-new-nginx', 'docker_image:nginx', 'image_name:nginx']),
+            ('docker.mem.rss', ['container_name:test-new-redis-latest', 'docker_image:redis:latest', 'image_name:redis', 'image_tag:latest']),
+            ('docker.mem.limit' ,['container_name:test-new-nginx', 'docker_image:nginx', 'image_name:nginx']),
+            ('docker.mem.in_use' ,['container_name:test-new-nginx', 'docker_image:nginx', 'image_name:nginx']),
+            # Container size metrics
+            ("docker.container.size_rootfs", ['container_name:test-new-nginx', 'docker_image:nginx', 'image_name:nginx']),
+            ("docker.container.size_rootfs", ['container_name:test-new-redis-latest', 'docker_image:redis:latest', 'image_name:redis', 'image_tag:latest']),
+            ("docker.container.size_rw", ['container_name:test-new-nginx', 'docker_image:nginx', 'image_name:nginx']),
+
+        ]
+
+        config = {
+            "init_config": {},
+            "instances": [{
+                "url": "unix://var/run/docker.sock",
+                "collect_container_size": True,
+                "collect_image_size": True,
+            },
+            ],
+        }
+
+        self.run_check(config, force_reload=True)
+        for mname, tags in expected_metrics:
+            self.assertMetric(mname, tags=tags, count=1, at_least=1)
```

Comment:
We can't do coverage report as the test can't run on travis or circleci. I run the test locally on a machine that have more containers than the one that are spawned.